sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -353,6 +367,30 @@ def create_or_update(
         name, version = component._get_rest_name_version()
         rest_component_resource = component._to_rest_object()
         result = None
+        try:
+            if not component._is_anonymous:
+                client_component_hash = component.get_component_hash(keys_to_omit=["version"])
+                rest_component_resource.properties.properties["client_component_hash"] = client_component_hash
+                remote_component_version = self.get_component_version(
+                    name=name
+                )  # will raise error if not found, so need set client_component_hash in the above line first.
+                remote_component_hash = None
+                if remote_component_version:
+                    remote_component_hash = remote_component_version.properties.properties.get("client_component_hash")
+                if remote_component_hash == client_component_hash and kwargs.get("skip_if_no_change"):
+                    component.version = remote_component_version.properties.component_spec.get(
+                        "version"
+                    )  # only update the default version component instead of creating a new version component
+                    version = component.version
+                    rest_component_resource = component._to_rest_object()
+                    logger.warning(
+                        "The component is not modified compared to the default version "
+                        "and the new version component registration is skipped."
+                    )
+                rest_component_resource.properties.properties["client_component_hash"] = client_component_hash
+        except Exception as e:
+            logger.error(f"Failed to compare and set client_component_hash, {e}")
+
```

Comment:
For v1.5 sdk, if found the registered component, it will update the description, display_name and tags of registered component.
https://msdata.visualstudio.com/Vienna/_git/AzureMlCli?path=/src/azure-ml-component/azure/ml/component/_api/_api.py&version=GBazureml-component/integration&line=1195&lineEnd=1198&lineStartColumn=25&lineEndColumn=82&lineStyle=plain&_a=contents

Reply:
Yes, because in V1.5, the parameter of `-- set xxx `is placed in `params_override`, independent of the component object, so need to update.
But in v2.0, currently the parameter` --set xxx` will be placed in the component object.
Anywhy, I always update the component, there will always be no errors, which is why I still continue to update the component even if the component hash no change.

Reply:
Got it. If so, do we need to add description, display_name and tags to omit_fields in calculating component hash?

Reply:
@c-w 
This depends on the design, whether the description or display_name or tags update requires the creation of a new version component.